lib/wcc/contentful/model_builder.rb

```diff
@@ -40,21 +40,31 @@ def build_model(typedef)
             new(raw, context) if raw.present?
           end
 
-          define_singleton_method(:find_all) do |context = nil|
-            raw = WCC::Contentful::Model.store.find_by(content_type: content_type)
-            raw.map { |r| new(r, context) }
+          define_singleton_method(:find_all) do |filter = nil, context = nil|
+            if filter
+              filter.transform_keys! { |k| k.to_s.camelize(:lower) }
+              bad_fields = filter.keys.reject { |k| fields.include?(k) }
+              raise ArgumentError, "These fields do not exist: #{bad_fields}" unless bad_fields.empty?
+            end
+
+            query = WCC::Contentful::Model.store.find_all(content_type: content_type)
+            query = query.apply(filter) if filter
+            query.map { |r| new(r, context) }
           end
 
           define_singleton_method(:find_by) do |filter, context = nil|
             filter.transform_keys! { |k| k.to_s.camelize(:lower) }
             bad_fields = filter.keys.reject { |k| fields.include?(k) }
             raise ArgumentError, "These fields do not exist: #{bad_fields}" unless bad_fields.empty?
 
-            query = WCC::Contentful::Model.store.find_by(content_type: content_type)
-            filter.each do |field, v|
-              query = query.eq(field, v, context)
-            end
-            query.map { |r| new(r, context) }
+            result = WCC::Contentful::Model.store.find_by(content_type: content_type, filter: filter)
+            new(result, context)
```

Comment:
breaking change - at least a minor version bump since we're still pre `1.0`

Reply:
Yep